pkg/provider/aws/secret.go

```diff
@@ -2,42 +2,63 @@ package aws
 
 import (
 	"fmt"
-
 	"github.com/klothoplatform/klotho/pkg/core"
-	"github.com/klothoplatform/klotho/pkg/multierr"
 	"github.com/klothoplatform/klotho/pkg/provider/aws/resources"
+	"github.com/pkg/errors"
 )
 
-func (a *AWS) GenerateSecretsResources(construct *core.Secrets, result *core.ConstructGraph, dag *core.ResourceGraph) error {
-	var merr multierr.Error
+func (a *AWS) expandSecrets(dag *core.ResourceGraph, construct *core.Secrets) error {
 	for _, secretName := range construct.Secrets {
-		merr.Append(a.generateSecret(construct, result, dag, secretName))
-	}
-	return merr.ErrOrNil()
-}
-
-func (a *AWS) generateSecret(construct core.Construct, result *core.ConstructGraph, dag *core.ResourceGraph, secretName string) error {
-	secret := resources.NewSecret(construct.Provenance(), secretName, a.Config.AppName)
-	dag.AddResource(secret)
-	a.MapResourceDirectlyToConstruct(secret, construct)
+		secretVersion, err := core.CreateResource[*resources.SecretVersion](dag, resources.SecretVersionCreateParams{
+			AppName:      a.Config.AppName,
+			Refs:         []core.AnnotationKey{construct.AnnotationKey},
+			Name:         secretName,
+			DetectedPath: secretName,
+		})
 
-	secretVersion := resources.NewSecretVersion(secret, secretName)
-	dag.AddDependenciesReflect(secretVersion)
-
-	for _, upstreamCons := range result.GetUpstreamConstructs(construct) {
-		unit, isUnit := upstreamCons.(*core.ExecutionUnit)
-		if !isUnit {
-			continue
+		if err != nil {
+			return err
 		}
 
-		actions := []string{`secretsmanager:DescribeSecret`, `secretsmanager:GetSecretValue`}
-		policyResources := []core.IaCValue{{
-			Resource: secret,
-			Property: resources.ARN_IAC_VALUE,
-		}}
-		policyDoc := resources.CreateAllowPolicyDocument(actions, policyResources)
-		policy := resources.NewIamInlinePolicy(fmt.Sprintf("%s-secretsmanager", secret.Name), []core.AnnotationKey{construct.Provenance()}, policyDoc)
-		a.PolicyGenerator.AddInlinePolicyToUnit(unit.Id(), policy)
+		mapped := false
+		for _, r := range dag.GetDownstreamResources(secretVersion) {
+			if secret, ok := r.(*resources.Secret); ok {
+				if err := a.MapResourceToConstruct(secret, construct); err != nil {
+					return err
+				}
+				mapped = true
+				break
+			}
+		}
+		if !mapped {
+			return fmt.Errorf("unable to find secret resource for secret version, %s", secretVersion.Id())
```

Comment:
how do we map multiple secrets to the same construct? i forgot to take that into account when i assumed it was 1:1 because our secrets capability is the only one which doesnt follow that.

Reply:
We iterate over the construct's `Secrets.Secrets` slic, create a `Secret` resource and populate the resource's `DetectedPath` value. Then we set each secret's `Path` field  then set the `Secret.Path` value as part of `configureResources()` to its `DetectedPath`. If the construct is `Config` instead of `Secrets`, we ignore the resource's `DetectedPath` field and instead retrieve it from klotho config.